apps/core/app/[locale]/(default)/account/[tab]/_components/modal.tsx

```diff
@@ -0,0 +1,67 @@
+'use client';
+
+import { Button } from '@bigcommerce/components/button';
+import {
+  Dialog,
+  DialogAction,
+  DialogCancel,
+  DialogContent,
+  DialogDescription,
+  DialogOverlay,
+  DialogPortal,
+  DialogTitle,
+  DialogTrigger,
+} from '@bigcommerce/components/dialog';
+import { X } from 'lucide-react';
+import { MouseEventHandler, PropsWithChildren } from 'react';
+
+interface Props extends PropsWithChildren {
+  actionHandler?: MouseEventHandler<HTMLButtonElement>;
+  title: string;
+  descriptionText?: string;
+  confirmationText?: string;
+  abortText?: string;
+}
+
+export const Modal = ({
+  abortText = 'Cancel',
+  actionHandler,
+  confirmationText = 'OK',
+  descriptionText,
+  title,
+  children,
+}: Props) => {
+  return (
+    <Dialog>
+      <DialogTrigger aria-controls="modal-content" asChild>
+        {children}
+      </DialogTrigger>
+      <DialogPortal>
+        <DialogOverlay />
+        <DialogContent id="modal-content">
+          <div className="flex justify-between gap-4 p-6">
+            <DialogTitle>{title}</DialogTitle>
+            <DialogCancel asChild>
+              <Button className="ms-auto w-min p-2" type="button" variant="subtle">
+                <X />
```

Comment:
This should probably get an accessible name
```suggestion
                <X>
                  <title>Close</title>
                </X>
```